src/node/waku_node.py

```diff
@@ -84,6 +109,46 @@ def start(self, **kwargs):
             logger.error(f"REST service did not become ready in time: {ex}")
             raise
 
+    @retry(stop=stop_after_delay(5), wait=wait_fixed(0.1), reraise=True)
+    def register_rln(self, **kwargs):
+        logger.debug("Registering RLN credentials...")
+        self._docker_manager.create_network()
+        self._ext_ip = self._docker_manager.generate_random_ext_ip()
+        self._ports = self._docker_manager.generate_ports()
+        self._rest_port = self._ports[0]
+        self._tcp_port = self._ports[1]
+        self._websocket_port = self._ports[2]
+        self._discv5_port = self._ports[3]
+        self._metrics_port = self._ports[4]
+        self._api = REST(self._rest_port)
+        self._volumes = []
+
+        default_args = {
+            "rln-creds-id": None,
+            "rln-creds-source": None,
+        }
+
+        for key, value in kwargs.items():
+            key = key.replace("_", "-")
+            default_args[key] = value
+
+        rln_args, rln_creds_set, keystore_path = self.parse_rln_credentials(default_args, True)
+
+        if rln_creds_set:
+            self._container = self._docker_manager.start_container(
+                self._docker_manager.image, self._ports, rln_args, self._log_path, self._ext_ip, self._volumes
+            )
+
+            logger.debug(f"Executed container from image {self._image_name}. REST: {self._rest_port} to register RLN")
+
+            delay(15)
```

Comment:
I would remove this hard sleep and 
- inside rln_credential_store_ready if file does not exist raise an exception to actually trigger the retry (right now there is nothing to trigger the retry because os.path.exists will return True or False but will not fail)
- increase the stop_after_delay(15) so it does a smart wait

Reply:
Retry would not help us here, because new registration would also create new file which needs to be written to disk. OS file cache is standing in our way.

Reply:
what happens in those 15 seconds that we need a sleep?
if it's a file created on disk then we can smart wait for that file to appear
if it's something new written inside of existing file we can get the state of the file before the operation and compare it each 500 ms to see if new stuff is written.
Anything but a 15 sec sleep that can be too much or too litlle

Reply:
I agree, smart wait is a way to go. I tried os.sync(), but this was too smart, no reaction.

Reply:
some custom function similar with rln_credential_store_ready (decorated with retry) should do the trick
and inside that function we can check if the file is created or whatever we need

Reply:
Fixed at https://github.com/waku-org/waku-interop-tests/pull/16/commits/5c168fd06be1f6a21d1d93c0465bbb20c7a66d9c